homeassistant/components/homewizard/switch.py

```diff
@@ -66,12 +69,22 @@ def __init__(
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        await self.coordinator.api.state_set(power_on=True)
-        await self.coordinator.async_refresh()
+        await self._async_set_power_on(True)
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
-        await self.coordinator.api.state_set(power_on=False)
+        await self._async_set_power_on(False)
+
+    async def _async_set_power_on(self, power_on: bool) -> None:
+        """Set power state helper method."""
+        try:
+            await self.coordinator.api.state_set(power_on=power_on)
+        except RequestError as ex:
+            raise HomeAssistantError from ex
+        except DisabledError as ex:
+            await self.hass.config_entries.async_reload(self.coordinator.entry_id)
+            raise HomeAssistantError from ex
```

Comment:
As all of these occurrences are the same, why not make it a decorator?

For example:

https://github.com/home-assistant/core/blob/9ed629d838ba4bbff87e8de9870a3d8875eb49a6/homeassistant/components/lametric/helpers.py#L22C2-L50

Reply:
O, seems nice 👍! Will pick that up asap.

Reply:
https://github.com/home-assistant/core/pull/84991/commits/a6cb39cf5da60d3d32f8b1cce28cc0e9fee741df should be it!